osf_tests/test_analytics.py

```diff
@@ -42,3 +44,132 @@ def test_increment_user_activity_counters(self):
+
+        page_counter = PageCounter.objects.get(_id=page_counter_id)
+        assert page_counter.total == 1
+        assert page_counter.unique == 1
+
+        PageCounter.update_counter(page_counter_id, {})
+
+        page_counter.refresh_from_db()
+        assert page_counter.total == 2
+        assert page_counter.unique == 1
+
+    @mock.patch('osf.models.analytics.session')
+    def test_download_update_counter_contributor(self, mock_session, user, project, file_node):
+        mock_session.data = {'auth_user_id': user._id}
+        page_counter_id = 'download:{}:{}'.format(project._id, file_node.id)
+
+        PageCounter.update_counter(page_counter_id, {'contributors': project.contributors})
+        page_counter = PageCounter.objects.get(_id=page_counter_id)
+        assert page_counter.total == 0
+        assert page_counter.unique == 0
+
+        PageCounter.update_counter(page_counter_id, {'contributors': project.contributors})
+
+        page_counter.refresh_from_db()
+        assert page_counter.total == 0
+        assert page_counter.unique == 0
+
+    def test_get_all_downloads_on_date(self, page_counter, page_counter2):
+        """
+        This method tests that multiple pagecounter objects have their download totals summed properly.
+
+        :param page_counter: represents a page_counter for a file node being downloaded
+        :param page_counter2: represents a page_counter for another file node being downloaded
+        """
+
+        date = datetime(2018, 2, 4)
+
+        total_downloads = PageCounter.get_all_downloads_on_date(date)
+
+        assert total_downloads == 45
+
+    def test_get_all_downloads_on_date_exclude_versions(self, page_counter, page_counter2, page_counter_for_individual_version):
+        """
+        This method tests that individual version counts for file node's aren't "double counted" in the totals
+        for a page counter. We don't add the file node's total to the versions total.
+
+        :param page_counter: represents a page_counter for a file node being downloaded
+        :param page_counter2: represents a page_counter for another file node being downloaded
+        """
+        date = datetime(2018, 2, 4)
+
+        total_downloads = PageCounter.get_all_downloads_on_date(date)
+
+        assert total_downloads == 45
+
+
+class TestPageCounterRegex:
+
+    def test_download_all_versions_regex(self):
+        # Checks regex to ensure we don't double count versions totals for that file node.
+
+        match = re.match(PageCounter.DOWNLOAD_ALL_VERSIONS_ID_PATTERN, 'bad id')
+        assert not match
+
+        match = re.match(PageCounter.DOWNLOAD_ALL_VERSIONS_ID_PATTERN, 'views:guid1:fileid')
+        assert not match
+
+        match = re.match(PageCounter.DOWNLOAD_ALL_VERSIONS_ID_PATTERN, 'download:guid1:fileid:0')
+        assert not match
+
+        match = re.match(PageCounter.DOWNLOAD_ALL_VERSIONS_ID_PATTERN, 'download:guid1:fileid')
+        assert match
+
```

Comment:
Nit: Remove blank lines at ends of files.

This didn't get caught by flake8 because we're ignoring tests (for now).